src/Car/CarRepository.php

```diff
@@ -0,0 +1,215 @@
+         inner join modele USING(id_modele) 
+         inner join puissance USING(id_puissance) 
+         inner join finition USING(id_finition)')->fetchAll(PDO::FETCH_OBJ);
+
+        $voitures = [];
+        foreach ($rows as $row) {
+            $voiture = new Car();
+            $voiture
+                ->setId($row->id_voiture)
+                ->setImmat($row->immat)
+                ->setDateImmat(new \DateTimeImmutable($row->date_immat))
+                ->setMarque($row->nom_marque)
+                ->setModele($row->nom_modele)
+                ->setPuissance($row->puissance_ch)
+                ->setFinition($row->nom_finition)
+                ->setImage($row->lien_img)
+                ->setPrix($row->prix);
+
+            $voitures[] = $voiture;
+        }
+
+        return $voitures;
+    }
+
+
+    public function create($immat, $date_immat, $id_marque, $id_modele, $id_puissance, $id_finition, $lien_img, $prix)
+    {
+        $date_immat = new \DateTimeImmutable($date_immat);
+
+        $statement = $this->connection->prepare("INSERT INTO \"voiture\" (immat,date_immat,id_marque,id_modele,id_puissance,id_finition,lien_img,prix)\
+                                                values(:immat,:date_immat,:id_marque,:id_modele,:id_puissance,:id_finition,:lien_img,:prix)");
+
+        $statement->bindParam(":immat", $immat);
+        $statement->bindParam(":date_immat", $date_immat);
+        $statement->bindParam(":id_marque", $id_marque);
+        $statement->bindParam(":id_puissance", $id_puissance);
+        $statement->bindParam(":id_finition", $id_finition);
+        $statement->bindParam(":lien_img", $lien_img);
+        $statement->bindParam(":prix", $prix);
+
+        $statement->execute();
+    }
+
+    public function delete($id_voiture) {
+        $statement = $this->connection->prepare("DELETE FROM \"location\" WHERE id_voiture = :id_voiture");
+        $statement->bindParam(":id_voiture", $id_voiture);
+        $statement->execute();
+    }
+
+    public function update($id_voiture,$modifications) {
+        foreach($modifications as $titre => $modif) {
+            $modifsPrepared = $modifsPrepared . $titre . " = :" . $titre . " ";
+        }
+        $statement = $this->connection->prepare("UPDATE FROM \"voiture\" SET " . $modifsPrepared . "WHERE id_voiture = :id_voiture");
+        foreach($modifications as $titre => $modif) {
+            $statement->bindParam(":" . $titre, $modif);
+        }
+        
+        $statement->execute();
+    }
+
+    public function searchBrand($brand) {
+        $statement = $this->connection->prepare("SELECT * FROM marque WHERE nom_marque LIKE ?");
+
+        $statement->execute(array("%".$brand."%"));
+        $rows = $statement->fetchAll(PDO::FETCH_OBJ);
+
+        $marques = [];
+        
+        foreach ($rows as $row) {
+            $marques[] = array($row->id_marque=>$row->nom_marque);
```

Comment:
La short array syntaxe est préférée dorénavant en PHP (et ça prend moins de place).

```suggestion
            $marques[] = [$row->id_marque=>$row->nom_marque];
```